app/app/Http/Controllers/PasswordRecoveryController.php

```diff
@@ -32,8 +33,49 @@ public function sendEmail(Request $request)
 		if ( !$matching_user ) {
 			return view('pages.password_recovery.unmatched_email');
 		}
-		
-		mail($email, 'Password Recovery', 'Your password has been reset.');
+		BaseUser::sendPasswordRecoveryEmail($matching_user);
 		return view('pages.password_recovery.email_sent');
 	}
-}
\ No newline at end of file
+
+	public function passworkRecover($user_email, $password_recovery_token) {
+		$matching_user = User::where('email', '=', $user_email)->first();
+
+		if ( !$matching_user ) {
+			return view('pages.password_recovery.unmatched_email');
+		}
+
+		if ( $matching_user->password_recovery_token == $password_recovery_token ) {
+			Session::put('PasswordRecovery',$matching_user->email);
+			return view('pages.password_recovery.reset_password');
+		}
+
+		return view('pages.password_recovery.unmatched_token');
+	}
+	public function resetPassword(Request $request) {
+		if ( Session::has('PasswordRecovery') ) {
+			$validation_rules = array(
+				'new_password'         => 'required',
+				'password_confirm'     => 'required|same:new_password'
+			);
+			$validator = Validator::make(Input::all(), $validation_rules);
+			$failing = $validator->fails();
+			if ($failing)
```

Comment:
Storing a value in a variable to use it only once seems like a waste.  It is more code with no benefit.

Remove the failing variable and substitute the expression like this:
```
if ( $validator->fails() )
```